cucumber-core/src/main/java/io/cucumber/core/runner/CachingGlue.java

```diff
@@ -160,6 +168,7 @@ public void addDefaultDataTableCellTransformer(
     @Override
     public void addDocStringType(DocStringTypeDefinition docStringType) {
         docStringTypeDefinitions.add(docStringType);
+        dirtyCache = true;
```

Comment:
To wit, doc strings are not involved in regex creation. So they shouldn't dirty the cache.